### ISSUE TITLE
[Enhancement] audit log print real sql in prepared statement

### ISSUE THREAD

**Issue by eldenmoon** [apache/doris#42553]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

1. Currently in audit log, when we use server side prepared statement, after we execute a query, the audit log will print the replaced prepared statement like `EXECUTE xxxx USING XXX`, but we want to improve it to print the real sql like `select * from tbl where  id = XXX` .The origin prepared command is like `select * from tbl where  id = ?`
2. In prepared  stage, the audit log status will be `NOOP` which is confusing, if we prepared success then we should return `OK`, you could reproduce this issue by running `./run-regression-test.sh --run -d prepared_stmt_p0` and grep `NOOP` in fe.audit.log

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by eldenmoon** [comment]
notice use `./run-regression-test.sh --run -d prepared_stmt_p0` can reproduce the first issue when using session variable like` set global enable_prepared_stmt_audit_log = true;`

**Comment by HexyinUESTC** [comment]
"I understand your point as follows: 
1. After setting set global `enable_prepared_stmt_audit_log = true;`,  we should use `select * from tbl where id = XXX` to replace `EXECUTE xxxx USING XXX`. 
2. When `set global enable_prepared_stmt_audit_log = false;`(default), the state of the audit log should be OK upon prepared success. (Here, I noticed that after a successful execution, the state is explicitly set to NOOP, and I’m unsure if this is intentional.)

Is my understanding correct? Below is the version I implemented based on this understanding:"
1. before 
![image](https://github.com/user-attachments/assets/0821aa29-a387-41ba-9cba-514ff9448276)
after
![image](https://github.com/user-attachments/assets/ecd4f293-e440-4697-ae56-3decd5cf8e4e)
2. before
```
State=NOOP|ErrorCode=0|ErrorMessage=|Time(ms)=0|ScanBytes=0|ScanRows=0|ReturnRows=0|StmtId=638|QueryId=***IsQuery=false|isNereids=true|feIp=******|StmtType=SELECT|Stmt=select * from tbl_prepared_stmt_in_list WHERE `k10` IN (?, ?, ?, ?, ?) order by 1|CpuTimeMS=0|
```
after
```
State=OK|ErrorCode=0|ErrorMessage=|Time(ms)=0|ScanBytes=0|ScanRows=0|ReturnRows=0|StmtId=319|QueryId=*****|IsQuery=false|isNereids=true|feIp=******|StmtType=SELECT|Stmt=select * from tbl_prepared_stmt_in_list WHERE `k10` IN (?, ?, ?, ?, ?) order by 1|CpuTimeMS=0|
```
Please let me know if I have misunderstood any part of this. Thank you.

**Comment by HexyinUESTC** [comment]
Should I add corresponding unit tests? I haven't found unit tests for this class. Or is it sufficient to just pass the regression tests?

**Comment by eldenmoon** [comment]
greate job， almost done . Another improement
1. Add a flag in audit log to indicate whether it is a prepared command example add a field named `CommandType=COM_QUERY` , `CommandType=COM_STMT_PREPARE`  to indicate the mysql command type(MysqlCommand.java). 
2. As you can see in audit log, there are many duplicated log with the same QueryId it's maybe a bug fix it

**Comment by eldenmoon** [comment]
it's difficult to add such regression-test, so pass the tests should be ok

**Comment by HexyinUESTC** [comment]
Thank you for your reply. 
1. It is easy to add this flag to all statements. I will complete this feature.
2. When I first reproduced the problem, I found that some SQL statements that should be recorded were not written to the audit, and some sentences are repeated many times. which confused me. However, I saw that the above task did not include this content, so I temporarily ignored this module. Should I open another issue to fix this?
